bot/exts/moderation/infraction/management.py

```diff
@@ -72,13 +73,17 @@ async def infraction_append(
 
         Use "p" or "permanent" to mark the infraction as permanent. Alternatively, an ISO 8601
         timestamp can be provided for the duration.
+
+        If a previous infraction reason does not end with an ending punctuation mark, this automatically
+        adds a period before the amended reason.
         """
-        await self.infraction_edit(
-            ctx=ctx,
-            infraction=infraction,
-            duration=duration,
-            reason=fr"{infraction['reason']} **\|\|** {reason}",
-        )
+        add_period = not END_PUNCTUATION_RE.match(infraction["reason"])
+
+        new_reason = "".join((
+            infraction["reason"], ". " if add_period else " ", reason,
+        ))
```

Comment:
Strange to see `join` used here. I find it confusing to follow at first glance, since I associate a `join` with doing something a bit more complicated typically. I suggest you simply use the `+` operator.